### PR TITLE
fix: reorder check authentication middleware

### PR DESCRIPTION
**Pull request by MounirDhahri**
The type of this PR is: **Bugfix**

This PR resolves [CX-690]

### Description
Fix issue with the user not getting logged out when their session is invalid

**PS:**
Placing the `checkAuthenticationMiddleware` after `principalFieldErrorMiddleware` fixed the issue 🎉

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[CX-434]: https://artsyproduct.atlassian.net/browse/CX-434
[CX-690]: https://artsyproduct.atlassian.net/browse/CX-690

### CODE REVIEW

File: src/lib/relay/middlewares/__tests__/checkAuthenticationMiddleware-tests.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { NativeModules } from "react-native"
+import { GraphQLResponseErrors, MiddlewareNextFn, RelayNetworkLayerResponse } from "react-relay-network-modern/node8"
+import { checkAuthenticationMiddleware } from "../checkAuthenticationMiddleware"
+import { GraphQLRequest } from "../types"
+
+describe(checkAuthenticationMiddleware, () => {
+  const middleware = checkAuthenticationMiddleware()
+
+  const request: GraphQLRequest = {
```
Comment (MounirDhahri) [inline src/lib/relay/middlewares/__tests__/checkAuthenticationMiddleware-tests.ts:9]: this is just to satisfy the compiler

File: src/lib/relay/middlewares/checkAuthenticationMiddleware.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { NativeModules } from "react-native"
 import { Middleware } from "react-relay-network-modern/node8"
 
-export const checkAuthenticationMiddleware: Middleware = (next) => async (req) => {
+export const checkAuthenticationMiddleware = (): Middleware => (next) => async (req) => {
```
Comment (pvinis) [inline src/lib/relay/middlewares/checkAuthenticationMiddleware.ts:4]: consistency 🥳

**Review by pvinis — APPROVED** [review]
Very nice! One thing I would suggest is to add a comment, something along the lines of "The auth middleware is waiting for the next middlewares to finish and then it runs its checks and logs the user out if needed. That's why we need to keep it as low (later in the middlewares array) as possible."

File: src/lib/relay/middlewares/checkAuthenticationMiddleware.ts (hunk 2 above)
Comment (MounirDhahri) [reply src/lib/relay/middlewares/checkAuthenticationMiddleware.ts:4]: I did it to make you happy 😄

**Review by ashfurrow — APPROVED** [review]
Yeah, looks good! I agree with Pavlos, a comment would be great. After that we can merge 👍

**Pull request merged**